### PR TITLE
wgsl: Add test for [[interpolate]] parameters

### PR DESCRIPTION
**Pull request by jrprice**
Tested on macOS with ToT Chromium.

<hr>

**Author checklist for test code/plans:**

- [X] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [X] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [X] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/shader/validation/shader_io/interpolate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  '[[interpolate(perspective, center)]]',
+  '[[interpolate(perspective, centroid)]]',
+  '[[interpolate(perspective, sample)]]',
+  '[[interpolate(linear)]]',
+  '[[interpolate(linear, center)]]',
+  '[[interpolate(linear, centroid)]]',
+  '[[interpolate(linear, sample)]]',
+] as const;
+
+g.test('parameters')
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/shader_io/interpolate.spec.ts:24]: This test name doesn't really seem to match the fact that this is testing parameter **interpolation decorations**.

File: src/webgpu/shader/validation/shader_io/interpolate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const code = generateShader({
+      attribute: '[[location(0)]]' + interpolate,
+      type: 'f32',
+      stage: t.params.stage,
+      io: t.params.io,
+      use_struct: t.params.use_struct,
+    });
+
+    t.expectCompileResult(
+      kValidInterpolationAttributes.some(x => x === interpolate),
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/shader_io/interpolate.spec.ts:57]: The CTS [uses `es2016`](https://github.com/gpuweb/cts/blob/59628599c4e4490228f1d7fdd4013be66d2a6946/tsconfig.json#L4) which has support for `Map` and `Set`s which would reduce the search complexity here.

File: src/webgpu/shader/validation/shader_io/interpolate.spec.ts (hunk 1 above)
Comment (jrprice) [reply src/webgpu/shader/validation/shader_io/interpolate.spec.ts]: Changed the name. The full path is now:
`shader,validation,shader_io,interpolate,type_and_sampling`

File: src/webgpu/shader/validation/shader_io/interpolate.spec.ts (hunk 2 above)
Comment (jrprice) [reply src/webgpu/shader/validation/shader_io/interpolate.spec.ts]: Done.

**Review by ben-clayton — COMMENTED** [review]
LGTM, @sarahM0 for approval

**Review by sarahM0 — APPROVED** [review]
LGTM. Let's merge this, I'll add the unique ids later.

**Pull request merged**